debezium-core/src/main/java/io/debezium/pipeline/signal/channels/FileSignalChannel.java

```diff
@@ -0,0 +1,141 @@
+    public static final Field SIGNAL_FILE = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "file")
+            .withDisplayName("Signal file name")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDescription("The name of the file for the signals to the connector")
+            .withValidation(Field::isRequired);
+    public static final String CHANNEL_NAME = "file";
+    private static final Logger LOGGER = LoggerFactory.getLogger(FileSignalChannel.class);
+    ObjectMapper mapper = new ObjectMapper();
+    private File signalFile;
+
+    @Override
+    public String name() {
+        return CHANNEL_NAME;
+    }
+
+    @Override
+    public void init(CommonConnectorConfig connectorConfig) {
+
+        Configuration signalConfig = connectorConfig.getConfig().subset(CONFIGURATION_FIELD_PREFIX_STRING, false)
+                .edit()
+                .withDefault(FileSignalChannel.SIGNAL_FILE, "file-signals.txt")
+                .build();
+        this.signalFile = new File(signalConfig.getString(SIGNAL_FILE));
+        LOGGER.info("Reading '{}' file for signals", signalFile.getAbsolutePath());
+    }
+
+    @Override
+    public List<SignalRecord> read() {
+        List<SignalRecord> signals = new ArrayList<>();
+
+        if (!signalFile.exists() || signalFile.isDirectory()) {
+            LOGGER.trace("Signal file not found '{}'", signalFile.getAbsolutePath());
+            return signals;
+        }
+
+        try {
+            // read all the signals from file. signal line must be json formatted
+            List<String> lines = Files.readAllLines(signalFile.toPath());
+            if (!lines.isEmpty()) {
+                // remove signals from file by overriding file with empty content
+                new FileWriter(signalFile, false).close();
+            }
+            Iterator<String> lineIterator = lines.iterator();
+            while (lineIterator.hasNext()) {
+                String signalLine = lineIterator.next();
+                if (signalLine == null || signalLine.isBlank()) {
+                    LOGGER.debug("Ignoring empty signal line: `{}`", signalLine);
+                    lineIterator.remove();
+                    continue;
+                }
+                try {
+                    SignalRecord signal = readSignalString(signalLine);
+                    signals.add(signal);
+                    LOGGER.info("Processing signal: {}, {}, {}, {}", signal.getId(), signal.getType(), signal.getData(), signal.getAdditionalData());
+                }
+                catch (final Exception e) {
+                    LOGGER.warn("Skipped signal due to an error '{}'", signalLine, e);
+                    try {
+                        Files.writeString(signalFile.toPath(), signalLine + System.lineSeparator(), StandardOpenOption.APPEND);
+                    }
```

Comment:
@jpechane @mfvitale Not sure if we should log it as warning/error and skip the signal or add it back to file? 

here if given line is not parsable or buggy it adds back to the file. which will create WARN message with each read

Reply:
I think that if the line in not well formed (not a valid JSON) you should warn and skip without put it back again. I don't know if there is a special error for which you need to put it again on the file.

Reply:
I agree, we do the same with db signals as well.